vaadin-messages-flow-parent/vaadin-messages-flow/src/main/java/com/vaadin/flow/component/messages/MessageInput.java

```diff
@@ -90,4 +95,35 @@ public Registration addSubmitListener(
             ComponentEventListener<SubmitEvent> listener) {
         return addListener(SubmitEvent.class, listener);
     }
+
+    /**
+     * Gets the internationalization object previously set for this component.
+     * <p>
+     * Note: updating the object content returned by this method will not update
+     * the component if not set back using
+     * {@link MessageInput#setI18n(MessageInputI18n)}.
+     *
+     * @return the i18n object, or {@code null} if one has not been set with
+     *         {@link #setI18n(MessageInputI18n)}
+     */
+    public MessageInputI18n getI18n() {
+        return i18n;
+    }
+
+    /**
+     * Sets the internationalization properties for this component. It enabled
+     * you to customize and translate the language used in the message input.
+     * <p>
+     * Note: updating the object properties after setting the i18n will not
+     * update the component. To make the changes effective, you need to set the
+     * updated object again.
+     *
+     * @param i18n
+     *            the i18n object, not {@code null}
+     */
+    public void setI18n(MessageInputI18n i18n) {
+        Objects.requireNonNull(i18n, "The i18n object should not be null");
```

Comment:
null is the default value for `this.i18n`. Wouldn't it be okay then to reset the situation with `setI18n(null)`?

Reply:
I agree that it would make sense, but
- The implementation would be a bit ugly. Although the value is `null` at server-side, the web component has a default value. So we should duplicate the default value in Java code to be able to reset the web component, and then we are in trouble if those get out of sync.
- This is consistent with other Java component i18n APIs.
- Adding support for setting `null` here is a non-breaking change, so adding it later is no issue.

So because the implementation is not trivial, I would prefer keeping all the i18n APIs consistent and change all of those together if/when someone asks for it and it becomes a priority.